pkg/apis/clusteroperator/v1alpha1/types.go

```diff
@@ -48,20 +48,37 @@ type ClusterList struct {
 }
 
 type ClusterSpec struct {
-	MasterNodes ClusterNodeGroup `json:"masterNodes"`
+	// MasterNodeGroup specificies the configuration of the master node group
+	MasterNodeGroup ClusterNodeGroup `json:"masterNodeGroup"`
 
+	// ComputeNodeGroups specify the configurations of the compute node groups
 	// +optional
-	ComputeNodeGroups []ClusterNodeGroup `json:"computeNodeGroups,omitempty"`
+	ComputeNodeGroups []ClusterComputeNodeGroup `json:"computeNodeGroups,omitempty"`
```

Comment:
When we dig in for a more formal defintion of the type, I'm curious if we should use the same type attached to the ClusterSpec and stored separately, or if we should resort to NodeGroupSpec's as well.

Reply:
I am not following what your two options are here. Would you care to elaborate?

Reply:
I think I misread, I see now that a NodeGroup has a NodeGroupSpec.